apps/web/components/booking/pages/BookingPage.tsx

```diff
@@ -147,7 +147,7 @@ const BookingPage = ({
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
-  const mutation = useMutation(createBooking, {
+  const createBooking = trpc.viewer.bookings.event.useMutation({
     onSuccess: async (responseData) => {
       const { uid, paymentUid } = responseData;
```

Comment:
Getting a type error for `PaymentUid` .

Reply:
1. You only need to narrow down the type before accessing paymentUid like `if("paymentUid" in responseData && responseData.paymentUid) { ... }` and it should work.
2. I'd recommend typing the CTX parameter as well as explicitly typing the return of the `handler` function in `bookings.tsx`, right now if you hover over responseData its type is hard to read.
3. I believe there's a bug in the code (not that it needs to be addressed in this PR, just wanted to mention it), if we require a payment and `handlePayment` throws because the payment app isn't implemented, we end up with no payment uid and `BookingPage.tsx` redirects the user to the successful booking page, I may be missing some business logic context here but I don't believe this is the intended behavior? @zomars or someone else could confirm this.

Anyways, if you want to keep the existing logic but get the types working I'd do something like:

```ts
// in bookings.tsx

type RequiredPaymentInfo = {
  message: string;
  // Safeguard against the potential bug I mentioned.
  paymentUid: Booking["uid"] | undefined;
};

// It'd be good to extract the createBooking function from the handler to make this type more
// dynamic
type NewBooking = Booking & {
  user: {
    timeZone: string;
    name: string | null;
    email: string;
  } | null;
  payment: Payment[];
  attendees: Attendee[];
} & { appsStatus?: AppsStatus[] };

type NewBookingWithRequiredPayment = NewBooking & RequiredPaymentInfo;

type ExistingBooking = Prisma.BookingGetPayload<{
  select: {
    uid: true;
    id: true;
    attendees: true;
    userId: true;
    references: true;
    startTime: true;
    user: true;
  };
}>;

type ExistingBookingWithRequiredPayment = ExistingBooking & RequiredPaymentInfo;

type BookingTypes =
  | NewBooking
  | NewBookingWithRequiredPayment
  | ExistingBooking
  | ExistingBookingWithRequiredPayment;

type CTX = {
  // You could look at the nextauth routes/trpc middlewares to make this type more accurate and move it somewhere
  // to be exported and used throughout the app, this type right here is just based on what the handler uses.
  user: Prisma.UserGetPayload<{ select: { id: true } }>;
}

async function handler(ctx: CTX, input: extendedBookingCreateBody): Promise<BookingTypes>{
  // ...
}



// in BookingPage.tsx

if("paymentUid" in responseData && responseData.paymentUid) {
  // ... responseData is now of type NewBookingWithRequiredPayment | ExistingBookingWithRequiredPayment
  return // ...
}

// ... responseData is now of type BookingTypes due to us having to check for the truthyness of paymentUid 
// (the potential bug I mentioned)
return // ...
```